File: LibraryWeb/src/main/java/view/GuestView.java
package view;

import com.fasterxml.jackson.databind.ObjectMapper;
import com.fasterxml.jackson.databind.node.*;

import javax.servlet.http.*;
import java.io.*;
import java.util.ArrayList;

/**
 * Created by Alex on 3/6/2018.
 */
public class GuestView extends HttpServlet{

    @Override
    public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
        response.setContentType("application/json; charset=UTF-8");
        ObjectMapper map = new ObjectMapper();
        ArrayNode array = map.createArrayNode();

        try(PrintWriter out = response.getWriter()) {
            String id = request.getAttribute("id").toString();
            String first = request.getAttribute("first").toString();
            String last = request.getAttribute("last").toString();

            ObjectNode guestNode = map.createObjectNode();
            guestNode.put("id", id);
            guestNode.put("first", first);
            guestNode.put("last", last);

            array.add(guestNode);
            out.write(array.toString());
        }
    }

    @Override
    public void doDelete(HttpServletRequest request, HttpServletResponse response) throws IOException {
        response.setContentType("application/json; charset=UTF-8");
        ObjectMapper map = new ObjectMapper();
        ArrayNode array = map.createArrayNode();

        try(PrintWriter out = response.getWriter()) {
            boolean check = Boolean.parseBoolean(request.getAttribute("check").toString());
            if (check) {
                String id = request.getAttribute("id").toString();
                String first = request.getAttribute("first").toString();
                String last = request.getAttribute("last").toString();
                ObjectNode guestNode = map.createObjectNode();
                guestNode.put("status", "deleted");
                guestNode.put("id", id);
                guestNode.put("first", first);
                guestNode.put("last", last);

                array.add(guestNode);
                out.write(array.toString());
            }else {
                ObjectNode error = map.createObjectNode();
                error.put("error", "Bad guest id");
                array.add(error);
                out.write(array.toString());
            }
        }
    }

    public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
        response.setContentType("application/json; charset=UTF-8");
        ObjectMapper map = new ObjectMapper();
        ArrayNode array = map.createArrayNode();

        try(PrintWriter out = response.getWriter()) {
            ArrayList<String> guestList = (ArrayList<String>)request.getAttribute("guests");
            for (int i = 0; i < guestList.size(); i += 3) {
                ObjectNode node = map.createObjectNode();
                node.put("id", guestList.get(i));
                node.put("first", guestList.get(i+1));
                node.put("last", guestList.get(i+2));
                array.add(node);
            }
            out.write(array.toString());
        }
    }
}

File: LibraryWebSpark/src/main/java/view/GuestView.kt
package view

import com.fasterxml.jackson.databind.ObjectMapper
import com.fasterxml.jackson.databind.node.ArrayNode
import com.fasterxml.jackson.databind.node.ObjectNode

/**
 * Created by Alex on 3/22/2018.
 */
object GuestView {

    fun getGuests(guests: ArrayList<String>, map: ObjectMapper): ArrayNode {
        val array = map.createArrayNode()
        var node: ObjectNode
        for (i in 0 until guests.size step 3) {
            node = map.createObjectNode()
            array.add( node.run {
                node.put("id", guests[i])
                node.put("first", guests[i+1])
                node.put("last", guests[i+2])
            } )
        }
        return array
    }

    fun addGuest(id: String, first: String, last: String, map: ObjectMapper): ObjectNode {
        val node = map.createObjectNode()
        return node.run {
            put("id", id)
            put("first", first)
            put("last", last)
        }
    }

    fun deleteGuest(id: String, first: String, last: String, map: ObjectMapper): ObjectNode {
        val node = map.createObjectNode()
        return node.run {
            put("status", "deleted")
            put("id", id)
            put("first", first)
            put("last", last)
        }
    }
}